Implemented empty constructor for TransformValue

njlawton opened the pull request:
Updated and added tests to validate the behavior of the constructor.

Current behavior
-Empty constructor and empty array in constructor produce the same object
-Empty constructor asMatrix method will give the 2D identity matrix as output
-Empty constructor cssString will give an empty string as output

Note: As of now the final behavior of an empty TransformValue has not been added to the typed om spec and so may need to change when this has been better defined.

Commit:
Implemented the TransformValue empty constructor
fixed the expected behaviour in the test file

Commit:
Added test case to TransformValue to check the behaviour of an empty TransformValue object

wilddamon commented on src/transform-value.js:
```diff
@@ -15,29 +15,13 @@
-
-    this._matrix = this._computeMatrix();
-    this.cssString = this._generateCssString();
+    return this._createSequenceTransform(values);
```
These functions don't return anything, so this looks a bit strange. Also there's a lot of indirection in the constructor that jumps all around the file now. It's also strange that no argument is treated differently to passing in an empty sequence. Maybe @shans should clarify what the expected behaviour for these cases is? It is not currently covered in the spec.

It would be clearer to keep the setting of _matrix and cssString in the constructor, but  modify the computeMatrix function to look like this:

``` javascript
  function TransformValue(values) {
    if (values === undefined) {
      values = [];
    }
    if (!Array.isArray(values)) {
      throw ...
    }
// Keep the old version of the constructor code
  }
...

  TransformValue.prototype._computeMatrix = function() {
    if (!this.transformComponents.length) {
      // return identity matrix
    }
    var matrix = this.transformComponents[0].asMatrix();
    for (var i = 1; i < this.transformComponents.length; ++i) {
      matrix = matrix.multiply(this.transformComponents[i].asMatrix());
    }
    return matrix;
  };
```

wilddamon commented on src/transform-value.js:
```diff
@@ -61,6 +45,30 @@
   };
 
+  TransformValue.prototype._createEmptyTransform = function() {
+    this._matrix = new Matrix(1, 1, 1, 1, 1, 1);
```
The identity matrix is 

``` javascript
new Matrix(1, 0, 0, 1, 0, 0);
```

Commit:
Improved structure of the code in the TransformValue constructor
Also fixed identity matrix (used to be 1,1,1,1,1,1 not 1,0,0,1,0,0)
Made empty arrays have the same affect as an empty constructor

rjwright commented on test/js/transform-value.js:
```diff
@@ -9,14 +9,21 @@ suite('TransformValue', function() {
   });
 
+  test('TransformValue empty constructor should create an object with ' + 
+    'the expected properties', function() {
```
Maybe say that it should represent a 2D identity matrix with an empty cssString?

rjwright commented:
Do we need this functionality?

LGTM w/ nit.

Commit:
Gave empty TransformValue test case a more detailed description
